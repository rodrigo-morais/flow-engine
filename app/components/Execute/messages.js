import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.ExecutePage.header',
    defaultMessage: 'Execute',
  },
  form: {
    object: {
      id: 'app.components.ExecutePage.form.object',
      defaultMessage: 'Object',
    },
    submit: {
      id: 'app.components.ExecutePage.form.submit',
      defaultMessage: 'Execute flow',
    },
  },
});
