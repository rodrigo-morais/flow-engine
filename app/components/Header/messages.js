import { defineMessages } from 'react-intl';

export default defineMessages({
  home: {
    id: 'boilerplate.components.Header.home',
    defaultMessage: 'Home',
  },
  register: {
    id: 'boilerplate.components.Header.register',
    defaultMessage: 'Register',
  },
  execute: {
    id: 'boilerplate.components.Header.execute',
    defaultMessage: 'Execute',
  },
});
