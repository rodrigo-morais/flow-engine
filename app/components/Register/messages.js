import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.RegisterPage.header',
    defaultMessage: 'Register',
  },
  form: {
    title: {
      id: 'app.components.RegisterPage.form.title',
      defaultMessage: 'Rule title',
    },
    id: {
      id: 'app.components.RegisterPage.form.id',
      defaultMessage: 'Rule id',
    },
    body: {
      id: 'app.components.RegisterPage.form.body',
      defaultMessage: 'Rule body',
    },
    passed: {
      id: 'app.components.RegisterPage.form.passed',
      defaultMessage: 'If rule passed',
    },
    failed: {
      id: 'app.components.RegisterPage.form.failed',
      defaultMessage: 'If rule failed',
    },
  },
});
