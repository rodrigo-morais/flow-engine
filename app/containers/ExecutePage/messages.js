import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.ExecutePage.header',
    defaultMessage: 'Execute',
  },
});
