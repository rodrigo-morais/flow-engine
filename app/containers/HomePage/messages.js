import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.HomePage.header',
    defaultMessage: 'Flow Engine',
  },
  description: {
    id: 'app.components.HomePage.description',
    defaultMessage: 'A flow engine is an application that executes a flow consisting of several linked rules, in this case against some incoming data',
  },
});
