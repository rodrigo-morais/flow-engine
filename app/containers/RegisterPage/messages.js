import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.RegisterPage.header',
    defaultMessage: 'Register',
  },
});
