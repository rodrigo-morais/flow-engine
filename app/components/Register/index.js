import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import H1 from 'components/shared/H1';
import Form from 'components/shared/Form';
import FormLine from 'components/shared/FormLine';
import messages from './messages';

const Register = ({ registerRule }) => {
  const lines = [
    {
      name: 'title',
      labelText: <FormattedMessage {...messages.form.title} />,
      inputType: 'text',
    },
    {
      name: 'id',
      labelText: <FormattedMessage {...messages.form.id} />,
      inputType: 'text',
    },
    {
      name: 'body',
      labelText: <FormattedMessage {...messages.form.body} />,
      inputType: 'textArea',
    },
    {
      name: 'passed',
      labelText: <FormattedMessage {...messages.form.passed} />,
      inputType: 'text',
    },
    {
      name: 'failed',
      labelText: <FormattedMessage {...messages.form.failed} />,
      inputType: 'text',
    },
  ];
  const formLines = () => lines.map((line, index) =>
    <FormLine
      key={index}
      name={line.name}
      labelText={line.labelText}
      inputType={line.inputType}
    />);

  const validateRule = (form) =>
    form.title.length > 0 && form.id.length > 0 && form.body.length > 0;

  const onSubmit = (event) => {
    event.preventDefault();

    const form = event.target;
    const rule = [].slice.call(form.querySelectorAll('input'))
      .map((field) => ({ [field.name]: field.value }))
      .reduce((previous, current) => ({ ...previous, ...current }), {});

    if (validateRule(rule)) {
      registerRule(rule);
    }

    form.reset();
  };

  return (
    <div>
      <H1>
        <FormattedMessage {...messages.header} />
      </H1>
      <Form onSubmit={onSubmit}>
        {formLines()}
        <button type="submit">Register</button>
      </Form>
    </div>
  );
};

Register.propTypes = {
  registerRule: PropTypes.func.isRequired,
};

export default Register;
