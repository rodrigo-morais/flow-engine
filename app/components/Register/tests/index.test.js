import React from 'react';
import { IntlProvider, FormattedMessage } from 'react-intl';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import Register from '../index';
import messages from '../messages';


describe('<Register />', () => {
  const registerRuleMock = jest.fn();

  it('renders according to design', () => {
    const component = renderer.create(
      <IntlProvider locale="en">
        <Register
          onRegisterRule={registerRuleMock}
        />
      </IntlProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the page title', () => {
    const renderedComponent = shallow(
      <Register
        onRegisterRule={registerRuleMock}
      />
    );
    expect(renderedComponent.contains(
      <FormattedMessage {...messages.header} />
    )).toEqual(true);
  });

  it('should contain 5 FormLines', () => {
    const renderedComponent = shallow(
      <Register
        onRegisterRule={registerRuleMock}
      />
    );
    const formLines = renderedComponent.find('FormLine');

    expect(formLines.length).toEqual(5);
  });

  it('should not call registerRuleMock when the form is submit empty', () => {
    const renderedComponent = mount(
      <IntlProvider locale="en">
        <Register
          onRegisterRule={registerRuleMock}
        />
      </IntlProvider>
    );

    const form = renderedComponent.find('form');
    form.reset = jest.fn();
    form.querySelectorAll = jest.fn().mockReturnValue([
      { name: 'title', value: '' },
      { name: 'id', value: '' },
      { name: 'body', value: '' },
    ]);

    const eventFake = {
      preventDefault: jest.fn(),
      target: form,
    };
    const submitButton = renderedComponent.find('button');

    submitButton.simulate('submit', eventFake);

    expect(registerRuleMock).not.toBeCalled();
  });

  it('should call registerRuleMock when the form is submit', () => {
    const renderedComponent = mount(
      <IntlProvider locale="en">
        <Register
          onRegisterRule={registerRuleMock}
        />
      </IntlProvider>
    );

    const rule = {
      title: 'title',
      id: 'id',
      body: 'body',
    };

    const form = renderedComponent.find('form');
    form.reset = jest.fn();
    form.querySelectorAll = jest.fn().mockReturnValue([
      { name: 'title', value: 'title' },
      { name: 'id', value: 'id' },
      { name: 'body', value: 'body' },
    ]);

    const eventFake = {
      preventDefault: jest.fn(),
      target: form,
    };
    const submitButton = renderedComponent.find('button');

    submitButton.simulate('submit', eventFake);

    expect(registerRuleMock).toBeCalledWith(rule);
  });
});
