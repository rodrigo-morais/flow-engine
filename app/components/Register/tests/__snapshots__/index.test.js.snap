exports[`<Register /> renders according to design 1`] = `
<div>
  <h1
    className="ccoFJN">
    <span>
      Register
    </span>
  </h1>
  <form
    className="eQdiUA"
    onSubmit={[Function]}>
    <fieldset
      className="gDzNqM">
      <label
        className="kEniqV"
        htmlFor="title"
        name="title">
        <span>
          Rule title
        </span>
      </label>
      <input
        className="LxCJS"
        name="title"
        type="text" />
    </fieldset>
    <fieldset
      className="gDzNqM">
      <label
        className="kEniqV"
        htmlFor="id"
        name="id">
        <span>
          Rule id
        </span>
      </label>
      <input
        className="LxCJS"
        name="id"
        type="text" />
    </fieldset>
    <fieldset
      className="gDzNqM">
      <label
        className="kEniqV"
        htmlFor="body"
        name="body">
        <span>
          Rule body
        </span>
      </label>
      <textarea
        className="eMewAs"
        cols="70"
        name="body"
        rows="10" />
    </fieldset>
    <fieldset
      className="gDzNqM">
      <label
        className="kEniqV"
        htmlFor="passed"
        name="passed">
        <span>
          If rule passed
        </span>
      </label>
      <input
        className="LxCJS"
        name="passed"
        type="text" />
    </fieldset>
    <fieldset
      className="gDzNqM">
      <label
        className="kEniqV"
        htmlFor="failed"
        name="failed">
        <span>
          If rule failed
        </span>
      </label>
      <input
        className="LxCJS"
        name="failed"
        type="text" />
    </fieldset>
    <button
      className="egBxff"
      type="submit">
      <span>
        Register
      </span>
    </button>
  </form>
</div>
`;
